### PR TITLE
Do not omit parens for 0-arity function definitions

### PR DESCRIPTION
**Pull request by uohzxela**

### CODE REVIEW

File: lib/ex_format.ex [hunk 1]
```diff
@@ -150,6 +150,16 @@ defmodule ExFormat do
     ast
   end
 
+  defp handle_zero_arity_fun(ast) do
+    case ast do
+      {sym, meta1, [{fun, meta2, nil} | rest]} when
+          sym in [:def, :defp, :defmacro, :defmacrop] ->
```
Comment (whatyouhide) [inline lib/ex_format.ex]: defdelegate should be here as well

**Review by whatyouhide — CHANGES_REQUESTED** [review]

**Comment by uohzxela** [comment]
@whatyouhide updated

File: lib/ex_format.ex [hunk 2]
```diff
@@ -150,6 +150,16 @@ defmodule ExFormat do
...
+    case ast do
+      {sym, meta1, [{fun, meta2, nil} | rest]} when
+          sym in [:def, :defp, :defmacro, :defmacrop, :defdelegate] ->
+        {sym, meta1, [{fun, meta2, []} | rest]}
+      _ ->
+        ast
+    end
+  end
+
   # TODO: rename to update_meta
```
Comment (lpil) [inline lib/ex_format.ex]: Seems this TODO can be removed

File: test/unit/fun_parens_test.exs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    test "for multiple function definitions in a block" do
+      bad = """
+      def f, do: something
+      defp f, do: something
+      defmacro f, do: something
+      defmacrop f, do: something
+      defdelegate f, do: something
+      """
+      good = """
+      def f(), do: something
```
Comment (lpil) [inline test/unit/fun_parens_test.exs:31]: Huh, I'm surprised by the params on zero arity function definitions. Wasn't aware that was a common or recommended style.

It's certainly not what `mix new` does

Comment (whatyouhide) [reply test/unit/fun_parens_test.exs:31]: @lpil for now ;)

The idea is: we call them with parens, we define them with parens when there are arguments, what sense does it make to define them without parens?

**Review by whatyouhide — APPROVED** [review]

File: test/unit/fun_parens_test.exs (hunk 3 above)
Comment (lpil) [reply test/unit/fun_parens_test.exs:31]: Seems reasonable. :)

**Pull request merged**